mne/encoding/feature.py

```diff
@@ -0,0 +1,269 @@
+"""Classes for transforming, preprocessing, and fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy.sparse import spmatrix, csr_matrix
+from ..utils import warn
+
+
+class FeatureDelayer(object):
```

Comment:
Ya I am not super set on any of the class names for this addition... @kingjr and I went through a couple different names and we landed on this, but definitely open to new ideas for what to call it. Could do `TimeLagsRepresentation` or `TimeLaggedRepresentation`, although that's a little wordy.